### PR TITLE
Update 'glfw' to v3.2.1

### PR DESCRIPTION
**Pull request by NukeBird**

### CODE REVIEW

File: cmake/configs/default.cmake [hunk 1]
```diff
@@ -178,7 +178,7 @@ hunter_config(geos VERSION 3.4.2)
 hunter_config(gflags VERSION 2.2.1)
 hunter_config(glbinding VERSION 2.1.3-p0)
 hunter_config(glew VERSION 2.0.0)
-hunter_config(glfw VERSION 3.3.0-p4)
+hunter_config(glfw VERSION 3.2.1)
```
Comment (ruslo) [inline cmake/configs/default.cmake]: I don't expect downgrading, please remove.

File: cmake/projects/glfw/hunter.cmake [hunk 2]
```diff
@@ -16,12 +16,23 @@ hunter_add_version(
...
+    PACKAGE_NAME
+    glfw
+    VERSION
     "3.2-p0"
     URL
     "https://github.com/hunter-packages/glfw/archive/3.2-p0.tar.gz"
     SHA1
     90f91bab3020db15a0fc07c27c53095fa2dbf1b3
-    )
+)
```
Comment (ruslo) [inline cmake/projects/glfw/hunter.cmake]: Please keep original formatting.

File: cmake/projects/glfw/hunter.cmake [hunk 3]
```diff
@@ -32,7 +43,7 @@ hunter_add_version(
     "https://github.com/hunter-packages/glfw/archive/3.3.0-p0.tar.gz"
     SHA1
     3621fc665397f07c57d1c77ac4d85d4634f18d3a
-    )
+)
```
Comment (ruslo) [inline cmake/projects/glfw/hunter.cmake]: Same, revert.

File: cmake/projects/glfw/hunter.cmake [hunk 4]
```diff
@@ -16,12 +16,23 @@ hunter_add_version(
     PACKAGE_NAME
     glfw
     VERSION
+    "3.2.1"
+    URL
+    "https://github.com/hunter-packages/glfw/archive/3.2.1.gz"
```
Comment (ruslo) [inline cmake/projects/glfw/hunter.cmake]: There is no such release.

Comment (NukeBird) [reply cmake/projects/glfw/hunter.cmake]: My bad, wrong repo: https://github.com/glfw/glfw/archive/3.2.1.tar.gz

File: cmake/configs/default.cmake (hunk 1 above)
Comment (NukeBird) [reply cmake/configs/default.cmake]: @ruslo it seems that "3.2.1" is latest release: https://github.com/glfw/glfw/releases

Or "3.3.0-p4" is a local release of hunter's fork? What shall I do?

Comment (ruslo) [reply cmake/configs/default.cmake]: > Or "3.3.0-p4" is a local release of hunter's fork?

Here is pull request with 3.3.0 version: https://github.com/hunter-packages/glfw/pull/1

@headupinclouds I can't see 3.3.0 tag. I guess 3.3.0 is next release from master, right?

File: cmake/projects/glfw/hunter.cmake (hunk 4 above)
Comment (ruslo) [reply cmake/projects/glfw/hunter.cmake]: > My bad, wrong repo: https://github.com/glfw/glfw

Will not work, this package need "hunterization":
* https://docs.hunter.sh/en/latest/creating-new/create/cmake-dependencies.html

That's why fork is used.

File: cmake/configs/default.cmake (hunk 1 above)
Comment (NukeBird) [reply cmake/configs/default.cmake]: So... I must somehow load glfw's release and rename its tag and release number?

Comment (ruslo) [reply cmake/configs/default.cmake]: > I must somehow load glfw's release and rename its tag and release number?

Not sure what you mean.

Also why do you need `3.2.1` release in the first place?
Official `3.2.1` release date is 18 Aug 2016: https://github.com/glfw/glfw/releases/tag/3.2.1
Fork `3.3.0` release date is 22 Aug **2017**: https://github.com/hunter-packages/glfw/releases/tag/3.3.0-p4

I expect all changed you need is already in `3.3.0` release.

Comment (NukeBird) [reply cmake/configs/default.cmake]: @ruslo they add gamepads and  SDL game controller database
https://github.com/glfw/glfw/issues/1117

Comment (ruslo) [reply cmake/configs/default.cmake]: What commit?

> https://github.com/glfw/glfw/blob/2867ca1e5b003b167fac33e8136a4698bb3aa074/include/GLFW/glfw3.h

Commit `2867ca1e5b003b167fac33e8136a4698bb3aa074` is in branch `master` and not released https://github.com/glfw/glfw/commit/2867ca1e5b003b167fac33e8136a4698bb3aa074

Comment (NukeBird) [reply cmake/configs/default.cmake]: @ruslo LOL I'm feeling stupid now :DDD

**Pull request closed without merging**